feat: Add debugger flag that can be turned on via RAY_DEBUG=legacy (#1312)
NVIDIA-NeMo/RL: docs/debugging.md

Reviewer: ```suggestion
This guide explains how to debug NeMo RL applications, covering two scenarios. It first outlines the procedure for debugging distributed Ray worker and actor processes using the Ray Distributed Debugger within a Slurm environment, and then details how to debug the main driver script.
```

![Ray Debugger Extension Step 4](./assets/ray-debug-step4.png)

> ## Debug with legacy Ray debugger

Reviewer: ```suggestion
## Debug with Legacy Ray Debugger
```

To use legacy ray debugger, you can use two ways

> 1. In general, set `RAY_DEBUG=legacy` and add `--ray-debugger-external` to your `ray start` command

Reviewer: ```suggestion
1. Set `RAY_DEBUG=legacy` and add `--ray-debugger-external` to your `ray start` command.
```

2. If you are using `ray.sub` in a slurm cluster, you can simply set `RAY_DEBUG=legacy` before `sbatch ray.sub`, the script can detect this environment variable and attach `--ray-debugger-external` automatically. 

> After you start ray with these changes, you can add `breakpoint` to your code. When you run the program, it will stop at where breakpoints are inserted. Then you can use a separate terminal to attach to the header node via `bash <JOB_ID>-attach.sh` (this script should automatically be generated by `ray.sub`), and run `ray debug` to see all the breakpoints. You can enter any breakpoint and interactively debug. Please refer to [Ray documentation](https://docs.ray.io/en/latest/ray-observability/user-guides/debug-apps/ray-debugging.html) for more info on this debugging approach.

Reviewer: ```suggestion
After you start Ray with these changes, you can add `breakpoint` to your code. When you run the program, it will pause where breakpoints are inserted. Then, in a separate terminal, attach to the head node using `bash <JOB_ID>-attach.sh` (this script should be automatically generated by `ray.sub`). From there, run `ray debug` to view all active breakpoints. You can enter any breakpoint and interactively debug. For more details on this debugging approach, refer to the [Ray documentation](https://docs.ray.io/en/latest/ray-observability/user-guides/debug-apps/ray-debugging.html).

```
